test/pylib/scylla_server.py

```diff
@@ -337,6 +337,27 @@ async def stop(self) -> None:
             self.control_connection = None
             self.control_cluster = None
 
+    async def stop_gracefully(self) -> None:
+        """Stop a running server. No-op if not running. Uses SIGTERM to
+        stop, so it is graceful. Waits for the process to exit before return."""
+        # Preserve for logging
+        hostname = self.hostname
+        logging.info("stopping server at host %s", hostname)
+        if not self.cmd:
+            return
+
+        try:
+            self.cmd.terminate()
+        except ProcessLookupError:
+            pass
+        else:
+            await self.cmd.wait()
+        finally:
+            if self.cmd:
+                logging.info("stopped server at host %s", hostname)
+            self.cmd = None
+            self.control_connection = None
```

Comment:
This needs to be updated to shutdown the control cluster (take a look at stop() in the current master.

Reply:
rebased